### ISSUE TITLE
Running depcruise through npx doesn't work as expected

### ISSUE THREAD

**Issue by mathiasose** [sverweij/dependency-cruiser#487]
### Expected Behavior
Running `depcruise` through `npx` to generate visualizations ought to work without needing to add it to package.json.

### Current Behavior
I get the correct visualization when I install `npm i dependency-cruiser@10.1.0` and then run `depcruise ...`, but not when I do `npx -p dependency-cruiser@10.1.0 depcruise ...`. The CLI tool executes without any apparent error, but the output is empty.

### Steps to Reproduce (for bugs)
I demonstrated in a fork of the repro-repo: https://github.com/mathiasose/dependency-cruiser-repro-repo/commit/f0edb037441cf9f12e8cba3fd59b2eaffbe20f2a

### Context
I tried to see if I could use this package as a drop-in one time visualization tool for a codebase without needing to add it to the package.json. This is usually possible with npx.

### Your Environment
- Version used: 10.1.0
- Node version: 14.17.0
- Operating System and version: Ubuntu 21.04

**Comment by sverweij** [comment]
hi @mathiasose - thanks for raising this. The behaviour you describe is what I expect dependency-cruiser to do. However, I do understand it's not immediately obvious why that would be.

### TL;DR

When running with npx dependency-cruiser might not have access to a typescript compiler and will restrict itself to javascript only.

### background - dependency-cruiser uses what is already there

dependency-cruiser does come shipped with a parser for javascript, but not with parsers for languages that compile to javascript. Instead it uses what is already available in the spot where it is installed - comparable to peer dependencies. This 
- guarantees it will use the correct version of these compilers, 
- ensures no unnecessary compilers take up space in your node_modules.
When you're on typescript odds are you're not interested in a coffee script, llivescript, vue or svelte compiler. Nor in one of the other two typescript compilers dependency-cruiser supports. 
- A nice side effect is that it keeps the installation of dependency-cruiser relatively small (~110kb zipped) - parsers for compile-to-javascript languages tend be quite large - e.g. 2Mb for coffeescript, 5Mb for babel (core), 8Mb for svelte, 23Mb for swc, 60Mb for typescript.

This also means that if you need to inspect typescript, a typescript compiler needs to be available on the spot the dependency-cruiser command runs from. There's a [FAQ entry](https://github.com/sverweij/dependency-cruiser/blob/develop/doc/faq.md#q-typescript-coffeescript-livescript-or-vue-single-file-component-sfc-dependencies-dont-show-up-how-can-i-fix-that) that explains this as well.

### running with npx

It turns out that with npx it is a bit of a gamble which binary it's running. It could be one of (1) your local folder (when something that matches the command happens to be there) (2) the spot where npm stores global installs (idem) (3) a location I haven't been able to determine yet (when you run npx with the `--package` option). For the first two situations the solution is obvious: install the compiler you need to that spot. For the last one I don't have a solution.

### why is dependency-cruiser not throwing a fit when there's no typescript compiler available?

When there's no compiler available it executes just fine, but with javascript only. There's currently not a way for it to tell  this is not as desired. It could grow some detection logic to gauge whether it's in a typescript/ vue/ coffeescript/ svelte environment and warn when there's a mismatch with available compilers. That'd complicate the tool a bit, though - and it would make everyone's runs slower than necessary. It is on the wishlist for the `--init` command, though. 

### next steps

I frankly didn't think to document the behaviour with npx. Your question makes me realise this needs to be corrected - I'll at least add an FAQ entry for it.

**Comment by mathiasose** [comment]
Thanks for the detailed answer. That means I can actually do it like https://github.com/mathiasose/dependency-cruiser-repro-repo/commit/1e7c382114b592c4736fcadb207f5da60be9b257 and get the desired output. Just need to be a bit careful that you match the typescript version in the npx command with the version you use in the application.

Adding this to the FAQ would be great. Would it also be possible to add some kind of warning message that prints when the CLI runs and it sees .ts files but no typescript compiler installed?

**Comment by sverweij** [comment]
hey @mathiasose I didn't know it was possible to  multiple -p's with `npx` - thanks for that! I've  added a small thing to the FAQ on npx and pasted your example in there.
